grammar.js

```diff
@@ -940,6 +972,8 @@ module.exports = grammar({
 
     rest_assignment: $ => prec(-1, seq('*', optional($._lhs))),
 
+    _fid: $ => choice(alias($.identifier_suffix, $.identifier), alias($.constant_suffix, $.constant)),
+    _fid_call: $ => prec.right(field('method', $._fid)),
```

Comment:
Could you explain this part of the change? I can't tell what the behavioral change is from introducing the `_fid` rule. 

I also wonder if there's some less abbreviated name that we could use instead of `_fid`; could it be written out as multiple words, similarly to how we use the full word version of most other names? What does the `f` stand for?

Reply:
Yes `fid` is a terrible name ;-) I got it from the Ruby's [parse.y](https://github.com/ruby/ruby/blob/master/parse.y#L9171-L9174) . I guess the `F` stands for function. These are identifiers ending with `?` or `!` which cannot be used as normal variable or constant names.

For example `Foo::Bar?` is a call to a method named `Bar?` with the module/class `Foo` as receiver, and not a scope resolution of the constant `Bar?` on the module/class `Foo`.

I'm currently working on a Ripper based AST printer that parses a Ruby file and prints it in the format expected by tree-sitter tests. That allows me to compare ASTs generated by the tree-sitter grammar to ones produced by the real Ruby grammar and find cases where tree-sitter and Ruby disagree.